<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>409</width>
    <height>492</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>409</width>
    <height>492</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>409</width>
    <height>492</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Options</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../../icons.qrc">
    <normaloff>:/icons/bezier.png</normaloff>:/icons/bezier.png</iconset>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="planarlinkage">
      <attribute name="title">
       <string>Planar Linkage</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QScrollArea" name="settingArea">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>359</width>
            <height>418</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <layout class="QFormLayout" name="formLayout_2">
               <item row="1" column="0">
                <widget class="QLabel" name="label_17">
                 <property name="text">
                  <string>Report in every: </string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_16">
                 <property name="text">
                  <string>Max Generation: </string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="report">
                 <property name="suffix">
                  <string>%</string>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>10</number>
                 </property>
                 <property name="value">
                  <number>1</number>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="maxGen">
                 <property name="minimum">
                  <number>1000</number>
                 </property>
                 <property name="maximum">
                  <number>10000</number>
                 </property>
                 <property name="singleStep">
                  <number>100</number>
                 </property>
                 <property name="value">
                  <number>1500</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="setDefault">
               <property name="text">
                <string>Reset to Default</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="limitPanel" native="true">
             <layout class="QGridLayout" name="gridLayout">
              <property name="leftMargin">
               <number>6</number>
              </property>
              <property name="topMargin">
               <number>6</number>
              </property>
              <property name="rightMargin">
               <number>6</number>
              </property>
              <property name="bottomMargin">
               <number>6</number>
              </property>
              <property name="spacing">
               <number>6</number>
              </property>
              <item row="2" column="2">
               <widget class="QDoubleSpinBox" name="AyMin">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>-50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="6" column="0">
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Connector: </string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QDoubleSpinBox" name="DxMin">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>-50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Dx: </string>
                </property>
               </widget>
              </item>
              <item row="6" column="2">
               <widget class="QDoubleSpinBox" name="LMin">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>5.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QDoubleSpinBox" name="DxMax">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Dy: </string>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>Max</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Ay: </string>
                </property>
               </widget>
              </item>
              <item row="4" column="2">
               <widget class="QDoubleSpinBox" name="DyMin">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>-50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="6" column="3">
               <widget class="QDoubleSpinBox" name="LMax">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="label_14">
                <property name="text">
                 <string>Min</string>
                </property>
               </widget>
              </item>
              <item row="8" column="0">
               <widget class="QLabel" name="label_15">
                <property name="text">
                 <string>Angle (degree):</string>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QDoubleSpinBox" name="AxMax">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="8" column="2">
               <widget class="QDoubleSpinBox" name="AMin">
                <property name="minimum">
                 <double>0.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>359.990000000000009</double>
                </property>
                <property name="value">
                 <double>0.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="8" column="3">
               <widget class="QDoubleSpinBox" name="AMax">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>360.000000000000000</double>
                </property>
                <property name="value">
                 <double>360.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QDoubleSpinBox" name="AxMin">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>-50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="4" column="3">
               <widget class="QDoubleSpinBox" name="DyMax">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Ax: </string>
                </property>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QDoubleSpinBox" name="AyMax">
                <property name="minimum">
                 <double>-999.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>Input:</string>
                </property>
               </widget>
              </item>
              <item row="5" column="2">
               <widget class="QDoubleSpinBox" name="IMin">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>5.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="5" column="3">
               <widget class="QDoubleSpinBox" name="IMax">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="7" column="0">
               <widget class="QLabel" name="label_19">
                <property name="text">
                 <string>Follower</string>
                </property>
               </widget>
              </item>
              <item row="7" column="2">
               <widget class="QDoubleSpinBox" name="FMin">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>5.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="7" column="3">
               <widget class="QDoubleSpinBox" name="FMax">
                <property name="minimum">
                 <double>0.010000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="value">
                 <double>50.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>27</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="algorithm">
      <attribute name="title">
       <string>Algorithm</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QTableWidget" name="APTable">
         <attribute name="horizontalHeaderDefaultSectionSize">
          <number>150</number>
         </attribute>
         <attribute name="horizontalHeaderMinimumSectionSize">
          <number>150</number>
         </attribute>
         <column>
          <property name="text">
           <string>Parameter</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Value</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
